src/com/puppetlabs/mq.clj

```diff
@@ -132,10 +132,16 @@
     (start-broker! (build-embedded-broker brokername dir config))
     (catch java.io.EOFException e
       (log/warn
-        (str "Caught EOFException on broker startup, trying to restart it "
-             "again to see if that solves it. This is probably due to "
-             "KahaDB corruption."))
-      (start-broker! (build-embedded-broker brokername dir config)))))
+       "Caught EOFException on broker startup, trying again."
+       "This is probably due to KahaDB corruption "
+       "(see \"KahaDB Corruption\" in the PuppetDB manual).")
+      (start-broker! (build-embedded-broker brokername dir config)))
+    (catch java.io.IOException e
+      (log/error
+       e "Unable to start broker in " (pr-str dir) "."
+       "This is probably due to KahaDB corruption or version downgrade "
```

Comment:
the version downgrade part is only applicable to one case here. I can't remember where we landed this morning on returning that for all IOExceptions. I wonder if getting rid of the "or version downgrade" and simply mentioning the issue in the upgrading notes would be sufficient.